### DISCUSSION TITLE
How to navigate programatically outside of components? (e.g. using redux)

### DISCUSSION THREAD

**Discussion by sem4phor** [expo/router#643] [Q&A] [answered]
Hello,
currently we are using `"redux-first-history": "^5.1.1",` to trigger navigations via redux. Is there a way to achieve this with expo-router. From what i saw the package only exposes hooks & components as mechanisms to navigate.

**Comment by wcastand** [comment, marked as answer]
https://blog.expo.dev/introducing-expo-router-v2-3850fd5c3ca1

that line should help with expo-router v2

```The new router export can be used as an imperative API to navigate from global stores and providers.```
